### ISSUE TITLE
🐛 Bug Report: "Connection refused" error on dart function

### ISSUE THREAD

**Issue by FabiKel** [appwrite/appwrite#6945]
### 👟 Reproduction steps

When I try to get any information from my database I get an `AppwriteException: null, Connection refused (0)`.

I tried it with dart runtime 2.17, 2.18 and 3.0. I tried in developement and production mode.
I deployed the functions with CLI 4.1.0

```
final client = Client()
      .setEndpoint('http://myEndpoint/v1')
      .setProject("myProjektID")
      .setKey(Platform.environment["API_KEY"])
      .setSelfSined();
      
final Databases databases = Databases(client);
await databases.get(databaseId: "myDatabaseID"); // error here
```

To test it, i gave the key every scope, but the error persists. I upgraded from 1.3.6 to 1.4.3 to 1.4.5. The error exists since the upgrade from 1.3 to 1.4.

### 👍 Expected behavior

Get the required data.

### 👎 Actual Behavior

Throws `AppwriteException: null, Connection refused (0)`

```
AppwriteException: null, Connection refused (0)
#0      ClientIO.call (package:dart_appwrite/src/client_io.dart:231)
<asynchronous suspension>
#1      Databases.get (package:dart_appwrite/services/databases.dart:71)
<asynchronous suspension>
#2      main (package:starter_template/main.dart:13)
<asynchronous suspension>
#3      Future.any.onValue (dart:async/future.dart:612)
<asynchronous suspension>
```

### 🎲 Appwrite version

Version 1.4.x

### 💻 Operating system

Linux

### 🧱 Your Environment

- tested with Appwrite 1.4.3 and 1.4.5
- tested with Dart Appwrite SDK 8.0.0 and 10.0.0
- tested with Dart runtime 2.17, 2.18, 3.0

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by adarsh-jha-dev** [comment]
Hey @FabiKel , i request you to please assign this to me

**Comment by stnguyen90** [comment]
@FabiKel, thanks for creating this issue! 🙏🏼 Where is your Appwrite instance deployed and are you using localhost for the endpoint? What is the endpoint?

**Comment by FabiKel** [comment]
The instance is on a server (vm) in the local network only.
 
```
_APP_ENV=development
_APP_DOMAIN=[HOSTNAME]
_APP_DOMAIN_FUNCTIONS=functions.localhost
_APP_DOMAIN_TARGET=[HOSTNAME]
```

In version 1.3 the same function was working. Since 1.4 it's not working anymore. Are there some other (new) variables I have to check?

**Comment by stnguyen90** [comment]
@FabiKel, perhaps the hostname or IP address you're using here changed:

```dart
setEndpoint('http://myEndpoint/v1')
```

**Comment by FabiKel** [comment]
I have found the error. 
Host name and IP address have not changed, then the client applications would also no longer work. 
But at the same time as updating from 1.3.6 to 1.4.3, I also brought Ubuntu up to the latest version. 
In the old setup, the host name as .env variable in the function was resolved to the private IP address of the server. In the current version, the host name is converted to 127.0.1.1. This caused the error. And I think this is caused by Ubuntu not Appwrite.

Thank you for your help.
